diff --git a/src/network/wait-online/manager.c b/src/network/wait-online/manager.c
index 6de3df9..1db12ff 100644
--- a/src/network/wait-online/manager.c
+++ b/src/network/wait-online/manager.c
@@ -42,10 +42,10 @@ static int manager_link_is_online(Manager *m, Link *l, LinkOperationalState s) {
                 return log_link_debug_errno(l, SYNTHETIC_ERRNO(EAGAIN),
                                             "link has not yet been processed by udev");
 
-        if (STR_IN_SET(l->state, "configuring", "pending"))
+/*        if (STR_IN_SET(l->state, "configuring", "pending"))
                 return log_link_debug_errno(l, SYNTHETIC_ERRNO(EAGAIN),
                                             "link is being processed by networkd");
-
+*/
         if (s < 0)
                 s = m->required_operstate >= 0 ? m->required_operstate : l->required_operstate;
 
